File: tests/libjournal/trash.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <p2/journal.h>
// support
#include <cassert>


// alias the type
using trash_t = pyre::journal::trash_t;


// exercise the trivial device
int main() {
    // instantiate
    trash_t trash;
    // check its name
    assert (trash.name() == "trash");

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/trash.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Verify that we can access the {trash} device
    """
    # get the pakcage
    import j2

    # instantiate and verify the name
    assert j2.trash().name == "trash"

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
